include/secp256k1.h

```diff
@@ -80,9 +80,34 @@ typedef struct {
     unsigned char data[64];
 } secp256k1_ecdsa_signature;
 
+/** Data structure that holds a sign-to-contract ("s2c") context. Sign-to-contract
+ *  allows a signer to commit to some data as part of a signature. If the nonce
+ *  function supports sign-to-contract, after creating it with
+ *  secp256k1_s2c_commit_context_create the context can be given to a signing
+ *  algorithm via the nonce data argument.
+ *
+ *  This structure is not opaque, but it is strongly discouraged to read or write to
+ *  it directly. Use the secp256k1_s2c_commit_* instead to access a sign-to-contract
+ *  context.
+ *
+ *  The exact representation of data inside is implementation defined and not
+ *  guaranteed to be portable between different platforms or versions. It is however
+ *  guaranteed to be 136 bytes in size, and can be safely copied/moved.
+ */
+typedef struct {
+    /* magic is set during initialization. It allows functions casting to
+     * s2c_commit_contexts from a void pointer to check if they actually got an
+     * s2c_commit_context and if it has been initialized. */
+    unsigned char magic[8];
+    unsigned char data[32];
+    unsigned char data_hash[32];
+    secp256k1_pubkey original_pubnonce;
```

Comment:
should the "public nonce" really be represented as `pubkey`? for me `ge` or `gej` would be more readable, but they're probably not exported....
(I got confused while diving into `secp256k1_ec_commit_seckey` and `secp256k1_ec_commit_tweak` about the whole public key thing)

Reply:
`ge` or `gej` are not exported. But, hm, perhaps this should be a pubnonce type if it already confuses people.

Reply:
Hm, yes. A public key is a public key, and a nonce is a different thing. (I mean we can't enforce anything in C but maybe it's helpful to use different names for the two public types at least.)

Do we have the same problem in other places too?

Reply:
On a second thought... if we really want to optimize for a simple API, then https://github.com/bitcoin-core/secp256k1/pull/589 is the way to go. There the entire problem is eliminated by hiding from the user that a signature contains a "nonce".

Reply:
We do have that there too https://github.com/bitcoin-core/secp256k1/pull/589/files?file-filters%5B%5D=.c&file-filters%5B%5D=.h&file-filters%5B%5D=.include#diff-222d6d707e263c79c462fb223b0e3ddcR98

Reply:
yes

Reply:
fair